src/sinks/aws_sqs.rs

```diff
@@ -0,0 +1,461 @@
+                attribute_names: None,
+                queue_url: self.queue_url.clone(),
+            })
+            .await
+            .map(|_| ())
+            .context(GetQueueAttributes)
+            .map_err(Into::into)
+    }
+
+    pub fn create_client(&self, resolver: Resolver) -> crate::Result<SqsClient> {
+        let region = (&self.region).try_into()?;
+        let client = rusoto::client(resolver)?;
+
+        let creds = rusoto::AwsCredentialsProvider::new(&region, self.assume_role.clone())?;
+
+        Ok(SqsClient::new_with(client, creds, region))
+    }
+}
+
+impl SqsSink {
+    pub fn new(
+        config: SqsSinkConfig,
+        cx: SinkContext,
+        client: SqsClient,
+    ) -> crate::Result<impl Sink<SinkItem = Event, SinkError = ()>> {
+        // https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-batch-api-actions.html
+        // Up to 10 events, not more than 256KB as total size.
+        let batch = BatchSettings::default()
+            .events(10)
+            .bytes(262_144)
+            .timeout(1)
+            .parse_config(config.batch)?;
+
+        let request = config.request.unwrap_with(&REQUEST_DEFAULTS);
+        let encoding = config.encoding.clone();
+        let fifo = config.queue_url.ends_with(".fifo");
+
+        let successful_messages = Arc::new(Mutex::new(HashSet::new()));
+
+        let sqs = SqsSink {
+            successful_messages: Arc::clone(&successful_messages),
+            client,
+            queue_url: config.queue_url,
+        };
+
+        let sink = request
+            .batch_sink(
+                SqsRetryLogic {
+                    successful_messages,
+                },
+                sqs,
+                VecBuffer::new(batch.size),
+                batch.timeout,
+                cx.acker(),
+            )
+            .sink_map_err(|e| error!("Fatal sqs sink error: {}", e))
+            .with_flat_map(move |e| iter_ok(encode_event(e, &encoding, fifo)));
+
+        Ok(sink)
+    }
+}
+
+impl Service<Vec<SendMessageBatchRequestEntry>> for SqsSink {
+    type Response = SendMessageBatchResult;
+    type Error = RusotoError<SendMessageBatchError>;
+    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, mut entries: Vec<SendMessageBatchRequestEntry>) -> Self::Future {
+        debug!(
+            message = "Sending records.",
+            events = %entries.len(),
+        );
+
+        let sizes: Vec<usize> = entries
+            .iter()
+            .map(|entry| entry.message_body.len())
+            .collect();
+
+        // Make sure that `id` in batch is uniq
+        if entries[0].id.len() < MESSAGE_ID_LENGTH {
+            for (i, entry) in entries.iter_mut().enumerate() {
+                // `unwrap` is safe because batch can sent only maximum 10 events
```

Comment:
… except that the batch event size specified above is a default, not a maximum, so it would be possible for the user to configure a batch event size larger than 10.